### PR TITLE
Add `fraction_digits` to `Time#iso8601`

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

In most cases, I think a timestamp representation needs milliseconds.
This change also adds a Goodcheck rule about the use of `Time#iso8601`.

See also <https://ruby-doc.org/stdlib-2.7.2/libdoc/time/rdoc/Time.html#method-i-iso8601>.

> Link related issues or pull requests.

None.

> Check the following.

- [ ] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: lib/runners/results.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def initialize(guid:)
       def as_json
         {
           guid: guid,
-          timestamp: timestamp.utc.iso8601
+          timestamp: timestamp.utc.iso8601(3)
```
Comment (ybiquitous) [inline lib/runners/results.rb:15]: > __[Goodcheck]__ Consider adding `fraction_digits` to when using `Time#iso8601`. ([view](https://sider.review/gh/repos/202486306/pulls/1915#issue-5289743))
>
> <details>
>
> In most cases, milliseconds are necessary. For example:
>
> ```ruby
> Time.now.iso8601    #=> "2021-01-08T11:35:05+09:00"
> Time.now.iso8601(3) #=> "2021-01-08T11:35:05.371+09:00"
> ```
>
> See also <https://ruby-doc.org/stdlib-2.7.2/libdoc/time/rdoc/Time.html#iso8601-method>.
>
> | Rule |
> | ---- |
> | `sider.runners.time_iso8601` |
>
> You can [close this issue](https://sider.review/gh/repos/202486306/pulls/1915#issue-5289743) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

以下の理由でこのIssueをクローズしました：
- When `fraction_digits` is given already.

Comment (ybiquitous) [inline lib/runners/results.rb:15]: > __[Goodcheck]__ Consider adding `fraction_digits` to when using `Time#iso8601`. ([view](https://sider.review/gh/repos/202486306/pulls/1915#issue-5289744))
>
> <details>
>
> In most cases, milliseconds are necessary. For example:
>
> ```ruby
> Time.now.iso8601    #=> "2021-01-08T11:35:05+09:00"
> Time.now.iso8601(3) #=> "2021-01-08T11:35:05.371+09:00"
> ```
>
> See also <https://ruby-doc.org/stdlib-2.7.2/libdoc/time/rdoc/Time.html#method-i-iso8601>.
>
> | Rule |
> | ---- |
> | `sider.runners.time_iso8601` |
>
> You can [close this issue](https://sider.review/gh/repos/202486306/pulls/1915#issue-5289744) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

以下の理由でこのIssueをクローズしました：
- When `fraction_digits` is given already.

**Comment by ybiquitous** [comment]
@yykamei I think this change does not affect Sider, but can you please review this just in case?

**Review by yykamei — APPROVED** [review]
LGTM

**Pull request merged**